### ISSUE TITLE
Dropdown "autoselection" does not select first instance in source

### ISSUE THREAD

**Issue by ottosson** [handsontable/handsontable#11333]
### Describe the bug

When typing in a "dropdown" cell you'd expect the first match to be selected/highlighted in the source dropdown menu. This is currently not the case. I haven't investigated exatly how it behaves but it feels a bit random sometimes.

### Video/Screenshots

In this video it seems to select in the order as if it was sorted alphabetically, but that's not always the case.

https://github.com/user-attachments/assets/44568f49-c503-4e58-99f7-6ca754db0819

### Provide a link to the demo with the bug reproduction

https://handsontable.com/demo

### Handsontable version

15.0.0

### Your environment

Webview2 (edge webview on windows)

**Comment by AMBudnik** [comment]
Hm... I do not know if I fully understand the request, @ottosson 

It seems that now (in v15.0.0) we have the same output as in the previous versions.

<img width="456" alt="Zrzut ekranu 2024-12-27 o 10 05 36" src="https://github.com/user-attachments/assets/bcef5d9a-3e65-4193-8309-5d3c5e0bcdc9" />

When a cell contains a value, that value is highlighted in the list of choices. If the cell is empty, the first value is highlighted instead.

ps. I cannot run the video. It seems to last only 0 seconds. Could you please resend that?

**Comment by ottosson** [comment]
Sorry about the video... don't know what went wrong. It was working when I published the issue...

Anyway!

Take a look at this screenshot from the demo page. I've typed "to" and would expect "Toronto" to be highlighted. But instead it's "Tokyo" which is alphabetically before "Toronto" but I would expect the best match first in the list to be selected.

![image](https://github.com/user-attachments/assets/3dc0ae8f-b990-456c-9741-230cee371f07)

Here's a fiddle as well: https://jsfiddle.net/4smpgyrt/

**Comment by AMBudnik** [comment]
Thank you for sharing, @ottosson I see what you mean.

The alphabetical order has always been consistent. Here's version [10.0.0](https://handsontable.com/docs/javascript-data-grid/changelog/#_10-0-0), for example: https://jsfiddle.net/asp4gjq3/2/. It behaves the same as v15.0.0. However, in both themes (Main and Horizon), there is no indication that 'Toronto' also matches the query. In a no-theme option, it is achieved by bolding 'To' in 'Toronto'. I don’t see this behavior specified in the requirements, so I’ll report it internally as a bug. Thank you for pointing it out.

**Comment by ottosson** [comment]
Ah! It makes a bit more sense now that I see the intended behavior!

But I would still expect it to select the first match in the list though... I don't know how the logic works behind the scene, but I would have expected it to be something like this to select the instance:
`list.find(x => x.toLowerCase().startsWith("to"))`

Maybe it's subjective...

**Comment by ottosson** [comment]
I don't mind seeing all alternatives as I type. So at least keep that as an option if nothing else. My experience is that users get confused when nothing shows up in the menu when a value is pre-selected.

**Comment by AMBudnik** [comment]
> when nothing shows up in the menu

**Comment by ottosson** [comment]
@AMBudnik, thanks that's great. But I still don't understand the order that the options are selected when writing/filtering.

Take this example:
![Image](https://github.com/user-attachments/assets/47f8c66a-df72-4dfe-bfdc-0418e06ec27f)

I would have expected the "AutoCAD PDF..." to be selected before "DWG to PDF".

It's not in the correct option order (the selected option is after the expected option in the list) and it's not the correct order if the selection is based on alphabetical sorting.

How does it determine what option is selected?

**Comment by AMBudnik** [comment]
Yes, you are right. I was testing `t` for `Tokyo` and `Toronto`and then (as expected) `Tokyo` is selected. However, for `m` it doesn't make sense at all as we have `Miami` selected. 

https://jsfiddle.net/k3suj4xg/ (v15.1.0)

![Image](https://github.com/user-attachments/assets/83fa8072-e7b7-4d9c-9dd8-a085a6370c5d)

I will create an issue for that and get back to you once we have any progress.

**Comment by ottosson** [comment]
Thanks!

**Comment by ottosson** [comment]
@AMBudnik Did you create a public issue or one for internal use only? I would love to be able to follow this issue.

**Comment by AMBudnik** [comment]
Hi @ottosson we have an internal issue for that. My colleague has already been assigned to fix it. I will do my best to let you know once he makes a pull request.

**Comment by AMBudnik** [comment]
Hi @ottosson We just released Handsontable v16.1.0, where the issue you reported is fixed.

Here https://handsontable.com/docs/react-data-grid/changelog/#_16-1-0 you can read more about the changes in this version. We also introduced a few new features. I hope you'll like it.

**Comment by ottosson** [comment]
Indeed it was! Thanks a lot!
